rust/cas_client/src/caching_client.rs

```diff
@@ -77,35 +72,34 @@ impl<T: Client + Debug + Sync + Send> Client for CachingClient<T> {
         hash: &MerkleHash,
         data: Vec<u8>,
         chunk_boundaries: Vec<u64>,
-    ) -> Result<(), CasClientError> {
+    ) -> Result<()> {
         // puts write through
         debug!(
             "CachingClient put to {}/{} of length {} bytes",
             prefix,
             hash,
             data.len()
         );
-        self.client.put(prefix, hash, data, chunk_boundaries).await
+        Ok(self
+            .client
+            .put(prefix, hash, data, chunk_boundaries)
+            .await?)
```

Comment:
remove the `Ok` and the `?` and if the error type doesn't match use a `map_err` preferably. This looks weirdly formatted, maybe just run a rustfmt instead?

Reply:
In this case, I used the ? syntax intentionally as the current xet_error class now logs the error, along with a backtrace, whenever the ? operator is used to convert between Error types. map_err doesn't have the logging part, so I'd prefer to avoid it wherever possible.

Reply:
might be fixable with ErrorPrinter trait then, not requirement to merge imo